packages/medusa/src/repositories/sales-channel.ts

```diff
@@ -1,5 +1,57 @@
 import { EntityRepository, Repository } from "typeorm"
 import { SalesChannel } from "../models/sales-channel"
+import { flatten, groupBy, map, merge } from "lodash"
+import { CustomFindOptions } from "../types/common"
 
 @EntityRepository(SalesChannel)
-export class SalesChannelRepository extends Repository<SalesChannel> {}
+export class SalesChannelRepository extends Repository<SalesChannel> {
+  public async findWithRelations(
+    relations: string[] = [],
+    optionsWithoutRelations: CustomFindOptions<
+      SalesChannel,
+      keyof SalesChannel
+    > = {}
+  ): Promise<SalesChannel[]> {
+    const entities = await this.find(optionsWithoutRelations)
+    const entitiesIds = entities.map(({ id }) => id)
+
+    const groupedRelations = {}
+    for (const rel of relations) {
+      const [topLevel] = rel.split(".")
+      if (groupedRelations[topLevel]) {
+        groupedRelations[topLevel].push(rel)
+      } else {
+        groupedRelations[topLevel] = [rel]
+      }
+    }
+
+    const entitiesIdsWithRelations = await Promise.all(
+      Object.entries(groupedRelations).map(([_, rels]) => {
+        return this.findByIds(entitiesIds, {
+          select: ["id"],
+          relations: rels as string[],
+        })
+      })
+    ).then(flatten)
+    const entitiesAndRelations = entitiesIdsWithRelations.concat(entities)
+
+    const entitiesAndRelationsById = groupBy(entitiesAndRelations, "id")
+    return map(entitiesAndRelationsById, (entityAndRelations) =>
+      merge({}, ...entityAndRelations)
+    )
```

Comment:
Entirely true, i was just seeing that we have common implementation in different repository that could be move later in a base repo. So i did the more powerful one instead of just a simple find

Reply:
Fine with both approaches. Since this can be copy-pasted from other repositories, we can fairly easily add it when we have the need.